### PR TITLE
meijerint_indefinite should produce a Piecewise result conditional upon all 'splitting point' assumptions made to obtain the ultimate antiderivative

### LINKED ISSUE #26477: Error in integral result using hyper
It seems that when computing the result of integrals that derive from the beta function, some unexpected factors appear :

```python
Python 3.11.7 (main, Dec 15 2023, 18:12:31) [GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sympy as sp
>>> sp.__version__
'1.12'
>>> x = sp.Symbol("x", positive=True)
>>> y = sp.Symbol("y", positive=True)
>>> z = sp.Symbol("z", real=True)
>>> i = sp.Integral(z**(x-1)*(1-z)**(y-1), (z, 0, 1)) # = beta(x, y) by definition
>>> i.doit()
gamma(x)*hyper((x, 1 - y), (x + 1,), 1)/gamma(x + 1)
>>> _.simplify()
gamma(x)*gamma(y)/gamma(x + y) # correct
>>> i = sp.Integral(z**(x+0.5-1)*(1-z)**(y-1), (z, 0, 1)) # = beta(x+0.5, y)
>>> i.doit().simplify()
(-1)**y*exp(I*pi*y)*gamma(y)*gamma(x + 0.5)/gamma(x + y + 0.5) # ???
>>> 
```

Notice that using a particular value for y can yield results that are not consistent with the previous calculation :
```python
>>> i = sp.Integral(z**(x-0.5)*sp.sqrt(1-z), (z, 0, 1)) # = beta(x+0.5, 1.5)
>>> i.doit().simplify()
-sqrt(pi)*gamma(x + 0.5)/(2*gamma(x + 2.0)) # only a sign error, recall gamma(3/2)=sqrt(pi)/2
```

### PR DESCRIPTION
**Pull request by ehren**
#### References to other Issues or PRs

Fixes #26477 

#### Release Notes

* integrals
  * meijerint_indefinite now produces a Piecewise result conditional upon all 'splitting point' assumptions made to obtain the ultimate antiderivative

### CODE REVIEW

**Comment by ehren** [comment]
Hello, the inadequate behavior of meijerint_indefinite pointed out by @haru-44 in #26477 touched upon some of the "pole problems" I was looking into a few years ago. I have a larger change to meijerint that actually addresses most of the pole problems right where they originate - but this is a can of worms (perhaps it should be discussed in another issue). I had the "simple" / "lookup1" / "f*g" case well handled and even a few more but I recall there were still issues when meijerint relies on mellin transforms (I have to look at some of my old un-pull-requested branches - and a few local patches - to determine the state of this wider work).

Anyhow, the matter of handling the particular special cases imposed on the result by the splitting points seems to be a matter independent of these pole conditions (I'm still leaning towards another set of changes I've got that address the remaining pole issues similar to the approach of heurisch rather than complicating the lookup rules used by meijerint further).

**Comment by ehren** [comment]
Just an aside comment to whether meijerint should be able to handle various special conditions itself (or whether the heurisch reintegration approach should be used) - I'm leaning towards the heurisch approach (though I should link to a sensible diff showing the "lookup table" changes needed if meijerint were to consider these simple poles). See the comments in the test case related to the results containing logarithms. Especially for the case Integral(1/x, x) == log(x) note this paper: https://algo.inria.fr/flajolet/Publications/FlGeSa05.pdf 

"The sequence f_n = log n is not holonomic."

(so, handwaving, it shouldn't be the result of a "hypergeometric (MeijerG) integrator"

**Comment by oscarbenjamin** [comment]
The failing pypy test is some sort of bug between pypy and pytest and is unrelated to changes here,

File: sympy/integrals/meijerint.py [hunk 1]
```diff
@@ -1664,27 +1664,53 @@ def meijerint_indefinite(f, x):
...
+            from sympy.simplify.radsimp import collect
+            def _collect_factor(e):
+                return collect(factor_terms(e), e.atoms(exp))
+            if isinstance(rv, Piecewise):
+                return rv.func(*[(_collect_factor(r), c) for r, c in rv.args])
+            return _collect_factor(rv)
+
+    def _compare(item):
+        r, c = item
+        if c is has_closed_form:
```
Comment (smichr) [inline sympy/integrals/meijerint.py:1702]: I'm not sure "is" should be used to compare SymPy objects. Perhaps "c == has_closed_form"?

**Comment by ehren** [comment]
Argh, my bad. This change "fixes" the issue but only because it changes the order of computed results (Piecewise fold then just happens to disregard the incorrect result). The changes to the sorting of results would be needed if meijerint_indefinite1 were ever to return more special cases corresponding to poles in the integration result e.g. with this change https://github.com/ehren/sympy/commit/3e3e67f7e431ee69c60c0e0e8a37916031efae0a (also a messier version with a few variations: https://github.com/sympy/sympy/compare/master...ehren:sympy:meijerint-special-case-lookup-with-mellin).

In the example from the linked issue, there are no special cases to be dealt with. Rather it seems the bug is in hyperexpand (just commenting out the hyperexpand call in meijerint indefinite seems to make both results numerically valid). I'll post the hyperexpand details in the issue and close this pull because it's not the proper fix.

**Pull request closed without merging**